File: app/src/main/java/com/denchic45/EventObserver.java
package com.denchic45;

import androidx.lifecycle.Observer;

public class EventObserver<T> implements Observer<T> {

    private final EventUnhandledContent<T> content;

    public EventObserver(EventUnhandledContent<T> content) {
        this.content = content;
    }

    @Override
    public void onChanged(T event) {
        if (event != null) {
            content.onEvent(event);
        }
    }

    public interface EventUnhandledContent<T> {
        void onEvent(T t);
    }
}

File: app/src/main/java/com/denchic45/EventObserver.kt
package com.denchic45

import androidx.lifecycle.Observer

class EventObserver<T>(private val content: EventUnhandledContent<T>) : Observer<T> {
    override fun onChanged(event: T) {
        if (event != null) {
            content.onEvent(event)
        }
    }

    fun interface EventUnhandledContent<T> {
        fun onEvent(t: T)
    }
}